argo/common/common_types_unit_test.go

```diff
@@ -0,0 +1,120 @@
+//go:build unit
+
+package common
+
+import (
+	"encoding/json"
+	
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Context("Marshal NamespacedName", Serial, func() {
+	name := RandomString()
+	namespace := RandomString()
+	namespacedName := NamespacedName{Namespace: namespace, Name: name}
+
+	When("value is provided", func() {
+		It("custom marshaller is called", func() {
+			serialised, err := json.Marshal(namespacedName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(serialised)).To(Equal(`"` + namespace + "/" + name + `"`))
+			deserialised := NamespacedName{}
+			Expect(json.Unmarshal(serialised, &deserialised)).To(Succeed())
+			Expect(deserialised).To(Equal(namespacedName))
+		})
+	})
+
+	When("pointer is provider", func() {
```

Comment:
Do you mean
```suggestion
	When("pointer is provided", func() {
```
?

Reply:
See https://github.com/sky-uk/kfp-operator/pull/352/files/e78e58f3ef3c6464dd3d9c2fcce26544d39f5512#diff-351efeff6045b8ad01ff61963051f1b236fe93ba9e21243be970e12f79bf7003L34 🐒

Reply:
Hahaha OK 😆 
still 🧐